File: java/src/hashing/LC_128_LongestConsectiveSequence.java
package hashing;

// https://leetcode.com/problems/longest-consecutive-sequence
import java.util.HashSet;

public class LC_128_LongestConsectiveSequence {

    public void lengthOfLongestConsectiveSequence(int[] arr) {

        HashSet<Integer> sequence = new HashSet<>();
        for (int num : arr) {
            sequence.add(num);
        }

        int longestSeqCount = 0, count = 0, currentNum = 0;
        for (Integer num : sequence) {
            if (!sequence.contains(num - 1)) {
                currentNum = num;
                count = 1;

                while (sequence.contains(currentNum + 1)) {
                    currentNum++;
                    count++;
                }

                if (count > longestSeqCount) {
                    longestSeqCount = count;
                }
            }
        }
        System.out.println(longestSeqCount);

    }

    public static void main(String[] args) {
        LC_128_LongestConsectiveSequence sol = new LC_128_LongestConsectiveSequence();
        sol.lengthOfLongestConsectiveSequence(new int[] { 100, 4, 200, 1, 3, 2 });
        sol.lengthOfLongestConsectiveSequence(new int[] { 0, 3, 7, 2, 5, 8, 4, 6, 0, 1 });
        sol.lengthOfLongestConsectiveSequence(new int[] { 1, 0, 1, 2 });
    }
}

File: python/hashing/LC_128_LongestConsectiveSequence.py
# https://leetcode.com/problems/longest-consecutive-sequence
from typing import List
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        maxLen = 0
        for num in nums_set:
            if num- 1 not in nums_set:
                current_num = num
                counter = 1
                while(current_num +1 in nums_set):
                    current_num += 1
                    counter += 1
                maxLen = max(maxLen, counter)
        return maxLen
    
sol = Solution()
assert sol.longestConsecutive([100,4,200,1,3,2]) == 4
assert sol.longestConsecutive([0,3,7,2,5,8,4,6,0,1]) == 9
assert sol.longestConsecutive([1,0,1,2]) == 3
assert sol.longestConsecutive([0,0,-1]) == 2

                    
                